install/docker/ubuntu-docker-install.md

```diff
@@ -0,0 +1,48 @@
+# Installing Docker for Kata Containers on Ubuntu
+
+> **Note:**
+>
+> - This guide assumes you have
+>   [already installed the Kata Containers packages](../ubuntu-installation-guide.md).
+
+1. Install the latest version of Docker with the following commands:
+
+   > **Note:** This step is only required in case Docker is not installed on the system.
+
+   ```bash
+   $ sudo -E apt-get -y install apt-transport-https ca-certificates software-properties-common
+   $ curl -sL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+   $ arch=$(dpkg --print-architecture)
+   $ sudo -E add-apt-repository "deb [arch=${arch}] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+   $ sudo -E apt-get update
+   $ sudo -E apt-get -y install docker-ce
+   ```
+
+   For more information on installing Docker please refer to the
+   [Docker Guide](https://docs.docker.com/engine/installation/linux/ubuntu).
+
+2. Configure Docker to use Kata Containers by default with the following commands:
+
+   ```bash
+   $ sudo mkdir -p /etc/systemd/system/docker.service.d/
```

Comment:
Now we can also refactor this config in `install/docker/docker-kata-setup.md` :smile:

Reply:
Yes, I was thinking about that as this block is common for all the distros. It's difficult to get the right balance though as we don't want to send the user off having to read too many additional files.

wdyd @grahamwhaley, @egernst?

Reply:
I agree we should keep the 'doc tree depth' shallow - and in this case, this is a command/sequence that changes infrequently, so I think it is probably best left replicated across the docs.
/me opens up an opportunity for @jodh-intel to mention that markdown is not the most extensive or flexible document format, and that I'm sure something like ReST has an `include` or similar fragment function we could use to do this better ;-)

Reply:
@jodh-intel and what about reference the users to the installation docs in docker and only document how to configure it to use kata as runtime. For would need to move the logic to install docker to kata-manager but we will have a more simple file.

Reply:
would that stop us automating it then though?

Reply:
the testing that is...

Reply:
To summarise,

1) we have human-readable install docs which contain machine-parseable command blocks.
2) we have a script that can "execute" these docs.
3) we are currently automatically "testing" all changes to the install guides in the CI using (2).

I believe we also have two potentially conflicting user requirements:

a) some users want to install by running a script ("I have no time - just install it already!")
b) some user would rather review the commands we are asking them to run as `root` (understandably ;) and run those commands manually (assuming there aren't too many).

Although we could throw away the install docs and `kata-manager` and replace them with a big `install-kata.sh`, I don't think that's a good idea, nor would it be palatable to a lot of users (group (b) above).

I vote we retain (1)-(3) and:

- add some words to all the install docs explaining that if you are on a non-critical box, you *could* run `kata-manager.sh ...` to avoid typing in all the commands below, but we don't recommend it.
- add a big fat warning about doing this as it might be dangerous.
- add a new `--dry-run`-type option to `kata-manager.sh` that will generate the script for `install-*` commands, but won't execute it. That would allow users to review the script before executing it (as `root`).

Reply:
I agree - ultimately the 'big fat script' action should be taken care of by distro packages - you install it through your distro package management thing, ideally from the distros repos, and it 'just works'(tm).

I think the `--dry-run` is going to be a useful alternative/addition - the docs can say at the top:
> **Note**: if you do not wish to copy or type all these instructions by hand, you can use the <blah blah> script to generate a runnable shell script from the - see <blah blah blah> for more details

Reply:
Hi @grahamwhaley - I've tweaked the wording a bit but now all the docs have the following standard wording as a Note at the top (ignore the bad formatting):

> - If you do not wish to copy or type all these instructions by hand, you can use either use the
>   [`kata-manager`](https://github.com/kata-containers/tests/blob/master/cmd/kata-manager/kata-manager.sh)
>   script to install the packaged system including your chosen container manager. Alternatively, you can generate
>   a runnable shell script from individual documents using the
>   [`kata-doc-to-script`](https://github.com/kata-containers/tests/blob/master/.ci/kata-doc-to-script.sh) script.